### ISSUE TITLE
[Domino Visual Editor] Design Properties for Forms

### ISSUE THREAD

**Issue by JoelProminic** [Moonshine-IDE/Moonshine-IDE#1215]
#1214 originally requested updates for both Forms and Views, but only Views were updated.  I created this separate issue for the Form side.

**Comment by JoelProminic** [comment]
Here is a DXL Example:
![image](https://github.com/Moonshine-IDE/Moonshine-IDE/assets/24532731/55383b6f-e54f-4d89-a65f-a6a4b80b6207)

```
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE database SYSTEM "xmlschemas/domino_11_0_1.dtd">
<database xmlns="http://www.lotus.com/dxl" version="11.0" maintenanceversion="1.0">
  <form name="TestFormProperties" hide="notes web v4" noreplace="true" propagatenoreplace="true" publicaccess="false" designerversion="8.5.3" renderpassthrough="true">
    <body>
      <richtext>
        <pardef id="1"/>
        <par def="1">Test</par>
      </richtext>
    </body>
    <item name="$$ScriptName" summary="false" sign="true">
      <text>TestFormProperties</text>
    </item>
  </form>
</database>
```

TODO:  return this in "note" format

**Comment by JoelProminic** [comment]
I confirmed this is not merged to master, and I don't remember where we left off on this.   I'm bumping this to the next release, and we can revisit after #1009 (so that we don't need to switch back and forth for the Bamboo builds).

**Comment by feather812002** [comment]
@JoelProminic 
I check this again in today , I merge master into this branch and to do some test 
![image](https://github.com/Moonshine-IDE/Moonshine-IDE/assets/8543391/3548bf7c-4cbb-4c0f-8c19-944eda74db4e)
And I test the $FLAG , it match your table list on above, 
So, it  already ready for merge to master in next.

**Comment by JoelProminic** [comment]
When revisiting the milestone, I thought this issue was part of the #1009 branch, but I see it is on branch `features/issue-1215-design-properties-for-form`.  I updated Bamboo to generate a test build

The build is currently failing with this.  I remember seeing a similar build error recently - I think master needs to be merged into this branch again:

    build	27-Mar-2024 18:08:39	    [mxmlc] /Users/mmb/bamboo-agent-home/xml-data/build-dir/MS-BMOB-JOB1/Moonshine-IDE/ide/MoonshineDESKTOPevolved/src/actionScripts/utils/EnvironmentSetupUtils.as(368): col: 36 Error: Access of possibly undefined property HAXE_SYMLINK_COMMANDS through a reference with static type Class.
    build	27-Mar-2024 18:08:39	    [mxmlc] 
    build	27-Mar-2024 18:08:39	    [mxmlc] 					setCommand += HelperConstants.HAXE_SYMLINK_COMMANDS.join(";") +";";
    build	27-Mar-2024 18:08:39	    [mxmlc] 					                              ^

**Comment by JoelProminic** [comment]
@feather812002 fixed the build.  I did some tests and confirmed:
- The new properties show up in the form UI (MainApplication)
- The properties are preserved if I close and reopen the form
- If I open the generated database in Designer, I see the properties are set correctly.  I tested a variety of combinations here, including the default values.

@feather812002, this should be ready for a merge into master.

**Comment by JoelProminic** [comment]
@feather812002, did this get merged?   I thought we had discussed this, but I see the issue didn't get updated.
